js/vapid.js

```diff
@@ -90,6 +71,30 @@ var vapid = {
         return reply;
     },
 
+    _array_to_str: function(array) {
+        /* convert a ByteArray into a string
+         */
+        return String.fromCharCode.apply(null, new Uint8Array(array));
+    },
+
+    urlsafe: function(str) {
+        return str
+            .replace(/\+/g, "-")
+            .replace(/\//g, "_");
+    },
+
+    url_btoa: function(data) {
```

Comment:
Heh, you are NOT alone, added bonus is that the official js calls (btoa() and atob()) actually just convert binary strings to base64. Good call, though, I'll see if I can come up with a more sane name for the wrapper.